src/main/java/gregtech/api/capability/impl/MultiblockRecipeLogic.java

```diff
@@ -103,6 +104,39 @@ protected IMultipleTankHandler getOutputTank() {
         return controller.getOutputFluidInventory();
     }
 
+    @Override
+    protected boolean canWorkWithInputs() {
+        MultiblockWithDisplayBase controller = (MultiblockWithDisplayBase) metaTileEntity;
+        if (controller instanceof RecipeMapMultiblockController) {
+            RecipeMapMultiblockController distinctController = (RecipeMapMultiblockController) controller;
+
+            if (distinctController.canBeDistinct() && distinctController.isDistinct()) {
+                boolean canWork = false;
+                if (invalidatedInputList.isEmpty()) {
+                    return true;
+                }
+                if (!metaTileEntity.getNotifiedFluidInputList().isEmpty()) {
+                    canWork = true;
+                    invalidatedInputList.clear();
+                    metaTileEntity.getNotifiedFluidInputList().clear();
+                }
+                Iterator<IItemHandlerModifiable> iterator = metaTileEntity.getNotifiedItemInputList().iterator();
+                while (iterator.hasNext()) {
+                    IItemHandlerModifiable bus = iterator.next();
+                    if (invalidatedInputList.remove(bus)) {
+                        canWork = true;
+                    }
+                    iterator.remove();
+                }
```

Comment:
This section of code could be put into an `else` for the `if` statement check on the notified fluids. Since if we have notified fluids, we already clear the invalidated list, meaning that the check in the while loop will never be true.